src/v/cluster/cluster_discovery.cc

```diff
@@ -331,83 +388,71 @@ ss::future<> cluster_discovery::discover_founding_brokers() {
     // Make sure we didn't exit the above calls without a reply.
     _as.check();
 
-    // At this point, assuming the seed replies are consistent with ours, we
-    // are a cluster founder.
-    ss::sstring error_msg;
-    brokers seed_brokers;
-    seed_brokers.reserve(config_seed_servers.size());
     if (!config::node().node_id().has_value()) {
         clusterlog.info("Using index based node ID {}", seed_idx);
         co_await ss::smp::invoke_on_all([&] {
             config::node().node_id.set_value(
               std::optional<model::node_id>(seed_idx));
         });
     }
+
+    // At this point, assuming the seed replies are consistent with ours, we
+    // are a cluster founder. Now verify consistency of replies, and populate
+    // the list of founding brokers and the node_uuid map.
+    brokers founding_brokers;
+    founding_brokers.reserve(config_seed_servers.size());
+    node_ids_by_uuid node_ids;
```

Comment:
nit: maybe call this `node_uuid_map` or `node_id_by_uuid` or something? Seems easy to conflate with `unique_node_ids`.